sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -2635,7 +2643,11 @@ object Sequence {
         var i = 0
 
         while (t < exclusiveItem ^ stepSign < 0) {
-          arr(i) = fromLong(t / scale)
+          arr(i) = if (scale == 1) {
+            fromLong(t / scale)
```

Comment:
I will change to `fromLong(t)` because it is same as ` fromLong(t / scale)` in `if` condition